src/utils/Auth.js

```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { getInstance } from 'd2/lib/d2';
+
+
+export const withAuth = (Component) => {
+
+    const Auth = (props, { d2 }) => {
+        const extraProps = {
+            getCurrentUser: () => d2.currentUser,
+            getModelDefinitionByName: (modelType) => d2.models[modelType]
+
+        }
+        return <Component {...props} {...extraProps} />
+    }
+    Auth.contextTypes = {
+        d2: PropTypes.object
+    }
+    return Auth;
+}
```

Comment:
It might be nice to add a displayName based in the original component for dev niceness. 
https ://reactjs.org/docs/react-component.html#displayname